website/docs/docs/build/metricflow-time-spine.md

```diff
@@ -290,6 +294,239 @@ and date_hour < dateadd(day, 30, current_timestamp())
+* We recommend having a time spine at the finest grain used in any of your dimensions to avoid unexpected errors. i.e., if you have dimensions at an hourly grain, you should have a time spine at an hourly grain.
+
+## Example time spine tables
+
+### Daily
+
+<VersionBlock lastVersion="1.6">
+<File name='time_spine_daily.sql'>
+
+```sql
+{{
+    config(
+        materialized = 'table',
+    )
+}}
+
+with days as (
+
+    {{
+        dbt_utils.date_spine(
+            'day',
+            "to_date('01/01/2000','mm/dd/yyyy')",
+            "to_date('01/01/2025','mm/dd/yyyy')"
+        )
+    }}
+
+),
+
+final as (
+    select cast(date_day as date) as date_day
+    from days
+)
+
+select * from final
+-- filter the time spine to a specific range
+where date_day > dateadd(year, -4, current_timestamp()) 
+and date_hour < dateadd(day, 30, current_timestamp())
+```
+</File>
+</VersionBlock>
+
+<VersionBlock firstVersion="1.7">
+<File name='time_spine_daily.sql'>
+
+
+```sql
+{{
+    config(
+        materialized = 'table',
+    )
+}}
+
+with days as (
+
+    {{
+        dbt.date_spine(
+            'day',
+            "to_date('01/01/2000','mm/dd/yyyy')",
+            "to_date('01/01/2025','mm/dd/yyyy')"
+        )
+    }}
+
+),
+
+final as (
+    select cast(date_day as date) as date_day
+    from days
+)
+
+select * from final
+where date_day > dateadd(year, -4, current_timestamp()) 
+and date_hour < dateadd(day, 30, current_timestamp())
+```
+
+</File>
+</VersionBlock>
+
+### Daily (BigQuery)
+
+Use this model if you're using BigQuery. BigQuery supports `DATE()` instead of `TO_DATE()`:
+<VersionBlock lastVersion="1.6">
+
+<File name="metricflow_time_spine.sql">
+
+```sql
+{{config(materialized='table')}}
+with days as (
+    {{dbt_utils.date_spine(
+        'day',
+        "DATE(2000,01,01)",
+        "DATE(2025,01,01)"
+    )
+    }}
+),
+
+final as (
+    select cast(date_day as date) as date_day
+    from days
+)
+
+select *
+from final
+-- filter the time spine to a specific range
+where date_day > dateadd(year, -4, current_timestamp()) 
+and date_hour < dateadd(day, 30, current_timestamp())
+```
+</File>
+</VersionBlock>
+
+<VersionBlock firstVersion="1.7">
+
+<File name="metricflow_time_spine.sql">
+
+```sql
+{{config(materialized='table')}}
+with days as (
+    {{dbt.date_spine(
+        'day',
+        "DATE(2000,01,01)",
+        "DATE(2025,01,01)"
+    )
+    }}
+),
+
+final as (
+    select cast(date_day as date) as date_day
+    from days
+)
+
+select *
+from final
+-- filter the time spine to a specific range
+where date_day > dateadd(year, -4, current_timestamp()) 
+and date_hour < dateadd(day, 30, current_timestamp())
+```
+
+</File>
+</VersionBlock>
+
+## Hourly time spine
+
+<File name='time_spine_hourly.sql'>
+
+```sql
+{{
+    config(
+        materialized = 'table',
+    )
+}}
+
+with hours as (
+
+    {{
+        dbt.date_spine(
```

Comment:
should this be dbt_utils.date_spine for 1.6 and lower? and dbt.date_spine for 1.7 and higher?

Reply:
Oooo I'm actually not sure on this one!